src/util/lazyWithRetry.tsx

```diff
@@ -0,0 +1,32 @@
+import { lazy } from "react";
+import { delay } from "./helpers";
+
+// This function is a wrapper around React.lazy that will retry the import
+// keeping the function type signature the same as React.lazy
+// using "any" type here allows for components with any type interface to be imported
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const lazyWithRetry = <T extends React.ComponentType<any>>(
+  importFunction: () => Promise<{ default: T }>,
+  retries = 3,
```

Comment:
@edlerd @Kxiru open to suggestions for reasonable retries and delay time. Note `delayTime` here essentially is the ceiling of how long we would wait to retry import again (for the final attempt)